server/types/callbacks.go

```diff
@@ -54,8 +54,9 @@ type ConnectionCallbacks struct {
 	OnConnected func(ctx context.Context, conn Connection)
 
 	// OnMessage is called when a message is received from the connection. Can happen
-	// only after OnConnected(). Must return a ServerToAgent message that will be sent
-	// as a response to the Agent.
+	// only after OnConnected(). 
+	// When the returned ServerToAgent message is nil, WebSocket will not send a 
+	// message to the Agent, and the HTTP request will respond to an empty message.
```

Comment:
I think we lost the bit that says if the return is not nil it will be sent as a response to the Agent. This now only talks about the nil return case.

Reply:
done.